pkg/rancher-components/src/components/Form/LabeledInput/LabeledInput.vue

```diff
@@ -90,24 +101,85 @@ export default (
     delay: {
       type:    Number,
       default: 0
-    }
+    },
+
+    placeholder: {
+      type:    [String, Number],
+      default: ''
+    },
+
+    placeholderKey: {
+      type:    String,
+      default: null
+    },
+
+    label: {
+      type:    String,
+      default: null
+    },
+
+    labelKey: {
+      type:    String,
+      default: null
+    },
+
+    value: {
+      type:    [String, Number, Object],
+      default: ''
+    },
+
+    mode: {
+      type:    String,
+      default: _EDIT,
+    },
+
+    rules: {
+      default:   () => [],
+      type:      Array,
+      // we only want functions in the rules array
+      validator: (rules: Array<unknown>) => rules.every((rule: unknown) => ['function'].includes(typeof rule))
+    },
+
+    required: {
+      type:    Boolean,
+      default: false,
+    },
+
+    disabled: {
+      type:    Boolean,
+      default: false,
+    },
+  },
```

Comment:
I think you mentioned this during our conversation. I don't think we want to have to define these props here or anywhere the composables are used. At the very least I think we should define the props in the composable files and allow the consumer to spread them in the object here. If there's a better solution we should do that instead.

Reply:
Yeah, I think it should be possible to do this. I was on the fence about implementing this right now, as I think that the PR is getting quite large. 

I don't think that taking this on will add too much complexity on top of what's represented here, so I'll give it a try.

Reply:
This was quite straightforward to perform and we still get prop types exposed in our templates. Let me know what you think!